MediaBrowser.Model/Plugins/PluginStatus.cs

```diff
@@ -0,0 +1,47 @@
+namespace MediaBrowser.Model.Plugins
+{
+    /// <summary>
+    /// Plugin load status.
+    /// </summary>
+    public enum PluginStatus
+    {
+        /// <summary>
+        /// This plugin requires a restart in order for it to load. This is a memory only status.
+        /// The actual status of the plugin after reload is present in the manifest.
+        /// eg. A disabled plugin will still be active until the next restart, and so will have a memory status of RestartRequired,
+        /// but a disk manifest status of Disabled.
+        /// </summary>
+        RestartRequired = 1,
+
+        /// <summary>
+        /// This plugin is currently running.
+        /// </summary>
+        Active = 0,
+
+        /// <summary>
+        /// This plugin has been marked as disabled.
+        /// </summary>
+        Disabled = -1,
+
+        /// <summary>
+        /// This plugin does not meet the TargetAbi / MaxAbi requirements.
+        /// </summary>
+        NotSupported = -2,
+
+        /// <summary>
+        /// This plugin caused an error when instantiated. (Either DI loop, or exception)
+        /// </summary>
+        Malfunction = -3,
+
+        /// <summary>
+        /// This plugin has been superceded by another version.
+        /// </summary>
+        Superceded = -4,
+
+        /// <summary>
+        /// An attempt to remove this plugin from disk will happen at every restart.
+        /// It will not be loaded, if unable to do so.
+        /// </summary>
+        DeleteOnStartup = -5
```

Comment:
Within just these three states there is an adjective, a noun, and a phrase. We should try to keep enums at least somewhat consistent when possible for clarity.

Reply:
Changed.